master/internal/rm/kubernetesrm/spec.go

```diff
@@ -349,6 +351,43 @@ func (p *pod) createPriorityClass(name string, priority int32) error {
 	return err
 }
 
+const maxChars int = 63
+const fmtAlphaNumeric string = "A-Za-z0-9"
+const fmtAllowedChars string = fmtAlphaNumeric + `\.\-_`
+
+var regDisallowedSpecialChars = regexp.MustCompile("[^" + fmtAllowedChars + "]")
+var regLeadingNonAlphaNumeric = regexp.MustCompile("^[^" + fmtAlphaNumeric + "]+")
+var regTrailingNonAlphaNumeric = regexp.MustCompile("[^" + fmtAlphaNumeric + "]+$")
+
+func validatePodLabelValue(value string) string {
+	errs := validation.IsValidLabelValue(value)
+	if len(errs) == 0 {
+		return value
+	}
+
+	// label value is not valid; attempt to fix it
+	// 0. convert dis-allowed special characters to underscore
+	fixedValue := regDisallowedSpecialChars.ReplaceAllString(value, "_")
+
+	// 1. strip leading non-alphanumeric characters
+	fixedValue = regLeadingNonAlphaNumeric.ReplaceAllString(fixedValue, "")
+
+	// 2. truncate to 63 characters
+	if len(fixedValue) > maxChars {
+		fixedValue = string(fixedValue[:maxChars])
+	}
+
+	// 3. strip ending non-alphanumeric characters
+	fixedValue = regTrailingNonAlphaNumeric.ReplaceAllString(fixedValue, "")
+
+	log.Warnf(
+		"conform to Kubernetes pod label value standards: reformatting %s to %s",
+		value, fixedValue,
+	)
```

Comment:
I'm torn on the log message; I think it could be helpful, but also spammy. Maybe switch to debug?

Reply:
+1 switch to debug